### PR TITLE
Initial ADR guidance

### PR DESCRIPTION
**Pull request by hal-eisen-MG**
[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: ARCHITECTURE_DECISION_RECORDS/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Architecture Decision Records
```
Comment (gitbuda) [inline ARCHITECTURE_DECISION_RECORDS/README.md:1]: Seems like a very good start 💪

I would place this README into `docs/adr/` or just `adr/` (to be aligned with the rest of the repo, not a strong opinion).

File: ARCHITECTURE_DECISION_RECORDS/README.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Each ADR will be assigned a monotonically increasing unique numeric
+identifier, which will be zero-padded to 3 digits. Each ADR will
+be in a single markdown file containing no more than one page of
+text, and the filename will start with that unique identifier,
+followed by a camel case phrase summarizing the problem.  For
+example: `001_architecture_decision_records.md` or
+`002_big_integration_cap_theorem.md`.
+
+We want to use an ADR when:
```
Comment (andrejtonev) [inline ARCHITECTURE_DECISION_RECORDS/README.md:29]: Who decides when to add an ADR?
My first instinct is to write an ADR for everything, but that will get polluted very fast.

Comment (hal-eisen-MG) [reply ARCHITECTURE_DECISION_RECORDS/README.md:29]: Good point - thank you!

I've added a clarification that the most senior engineer on the project will evaluate and decide based on the 4 criteria (Significant Impact, Long-Term Ramifications, Architectural Principles, Controversial Choices).

File: ARCHITECTURE_DECISION_RECORDS/README.md (hunk 1 above)
Comment (hal-eisen-MG) [reply ARCHITECTURE_DECISION_RECORDS/README.md]: Fair enough. I moved changed to use a top level folder called "ADRs".

**Pull request merged**